### PR TITLE
Generate manifests

### PR DESCRIPTION
**Pull request by fiunchinho**
Every time we want to upgrade our app to a new upstream release, we need to manually check if there have been changes in upstream manifests like webhooks, deployment, etc. This is a time consuming and error prone task.

With this approach, the helm manifests are generated from the manifests attached to the upstream release.

I tried to reduce the customizations to the minimum, only keeping those that are truly required.

### CODE REVIEW

**Comment by tfussell** [comment]
@fiunchinho I see that you're using the `capz-system` namespace. From our last conversation here https://gigantic.slack.com/archives/C02MS174K/p1640192687215500?thread_ts=1640191745.213500&cid=C02MS174K it sounded like we were leaning more towards using the `giantswarm` namespace only. I'm not necessarily opposed to it, but I'd like to reach a consensus before we move to deploying in multiple namespaces.

**Comment by fiunchinho** [comment]
> @fiunchinho I see that you're using the `capz-system` namespace. From our last conversation here https://gigantic.slack.com/archives/C02MS174K/p1640192687215500?thread_ts=1640191745.213500&cid=C02MS174K it sounded like we were leaning more towards using the `giantswarm` namespace only. I'm not necessarily opposed to it, but I'd like to reach a consensus before we move to deploying in multiple namespaces.

If you take a look at the generated manifests, [like the `Deployment` one](https://github.com/giantswarm/cluster-api-provider-azure-app/pull/42/files#diff-259196409651a21827eb262e0b49649ea082c3305d5ce7388fc1c5b70249f9aa), I'm setting the namespace to the Helm release namespace so that we could use the `giantswarm` namespace. Maybe I missed some? The manifests that mention the `capz-system` namespace are the ones that I use to match against, so that I can apply the patches/overlays.

**Comment by tfussell** [comment]
> If you take a look at the generated manifests, [like the `Deployment` one](https://github.com/giantswarm/cluster-api-provider-azure-app/pull/42/files#diff-259196409651a21827eb262e0b49649ea082c3305d5ce7388fc1c5b70249f9aa), I'm setting the namespace to the Helm release namespace so that we could use the `giantswarm` namespace. Maybe I missed some? The manifests that mention the `capz-system` namespace are the ones that I use to match against, so that I can apply the patches/overlays.

I have trouble determining exactly what the rendered manifest will be. No problem if you're still developing and testing this. I just came here from the CRD management issue to see what you had planned.

**Comment by fiunchinho** [comment]
This is the output of `helm template helm/cluster-api-provider-azure -n giantswarm --set Installation.V1.Guest.SSH.SSOPublicKey=abc`

<details><summary>helm output</summary>

```yaml
---
# Source: cluster-api-provider-azure/templates/v1_serviceaccount_capz-manager.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  labels:
    app.giantswarm.io/branch: '[[ .Branch ]]'
    app.giantswarm.io/commit: '[[ .SHA ]]'
    app.kubernetes.io/instance: 'RELEASE-NAME'
    app.kubernetes.io/managed-by: 'Helm'
    app.kubernetes.io/name: cluster-api-provider-azure
    app.kubernetes.io/version: '0.0.1'
    cluster.x-k8s.io/provider: infrastructure-azure
    helm.sh/chart: cluster-api-provider-azure
  name: capz-manager
  namespace: 'giantswarm'
---
# Source: cluster-api-provider-azure/templates/ssh-sso-public-key-secret.yaml
apiVersion: v1
kind: Secret
metadata:
  labels:
    role: "ssh-sso-public-key"
    cluster.x-k8s.io/provider: infrastructure-azure
    app.kubernetes.io/name: "cluster-api-provider-azure"
    app.kubernetes.io/instance: "RELEASE-NAME"
    app.giantswarm.io/branch: "[[ .Branch ]]"
    app.giantswarm.io/commit: "[[ .SHA ]]"
    app.kubernetes.io/managed-by: "Helm"
    app.kubernetes.io/version: "0.0.1"
    helm.sh/chart: "cluster-api-provider-azure-0.0.1"
  name: cluster-api-provider-azure-ssh-sso-pub-key
  namespace: giantswarm
data:
  value: "YWJj"
---
# Source: cluster-api-provider-azure/templates/rbac.authorization.k8s.io_v1_clusterrole_capz-manager-role.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  creationTimestamp: null
  labels:
    app.giantswarm.io/branch: '[[ .Branch ]]'
    app.giantswarm.io/commit: '[[ .SHA ]]'
    app.kubernetes.io/instance: 'RELEASE-NAME'
    app.kubernetes.io/managed-by: 'Helm'
    app.kubernetes.io/name: cluster-api-provider-azure
    app.kubernetes.io/version: '0.0.1'
    cluster.x-k8s.io/provider: infrastructure-azure
    helm.sh/chart: cluster-api-provider-azure
  name: capz-manager-role
rules:
- apiGroups:
  - ""
  resources:
  - events
  verbs:
  - create
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - ""
  resources:
  - secrets
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - aadpodidentity.k8s.io
  resources:
  - azureidentities
  - azureidentities/status
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - aadpodidentity.k8s.io
  resources:
  - azureidentitybindings
  - azureidentitybindings/status
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - cluster.x-k8s.io
  resources:
  - clusters
  - clusters/status
  verbs:
  - get
  - list
  - patch
  - watch
- apiGroups:
  - cluster.x-k8s.io
  resources:
  - machinepools
  - machinepools/status
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - cluster.x-k8s.io
  resources:
  - machines
  - machines/status
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - ""
  resources:
  - nodes
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - infrastructure.cluster.x-k8s.io
  resources:
  - azureclusteridentities
  - azureclusteridentities/status
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - infrastructure.cluster.x-k8s.io
  resources:
  - azureclusters
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - infrastructure.cluster.x-k8s.io
  resources:
  - azureclusters/status
  verbs:
  - get
  - patch
  - update
- apiGroups:
  - infrastructure.cluster.x-k8s.io
  resources:
  - azuremachinepoolmachines
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - infrastructure.cluster.x-k8s.io
  resources:
  - azuremachinepoolmachines/status
  verbs:
  - get
  - patch
  - update
- apiGroups:
  - infrastructure.cluster.x-k8s.io
  resources:
  - azuremachinepools
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - infrastructure.cluster.x-k8s.io
  resources:
  - azuremachinepools/status
  verbs:
  - get
  - patch
  - update
- apiGroups:
  - infrastructure.cluster.x-k8s.io
  resources:
  - azuremachines
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - infrastructure.cluster.x-k8s.io
  resources:
  - azuremachines/status
  verbs:
  - get
  - patch
  - update
- apiGroups:
  - infrastructure.cluster.x-k8s.io
  resources:
  - azuremachinetemplates
  - azuremachinetemplates/status
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - infrastructure.cluster.x-k8s.io
  resources:
  - azuremanagedclusters
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - infrastructure.cluster.x-k8s.io
  resources:
  - azuremanagedclusters/status
  verbs:
  - get
  - patch
  - update
- apiGroups:
  - infrastructure.cluster.x-k8s.io
  resources:
  - azuremanagedcontrolplanes
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - infrastructure.cluster.x-k8s.io
  resources:
  - azuremanagedcontrolplanes/status
  verbs:
  - get
  - patch
  - update
- apiGroups:
  - infrastructure.cluster.x-k8s.io
  resources:
  - azuremanagedmachinepools
  verbs:
  - create
  - delete
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - infrastructure.cluster.x-k8s.io
  resources:
  - azuremanagedmachinepools/status
  verbs:
  - get
  - patch
  - update
---
# Source: cluster-api-provider-azure/templates/rbac.authorization.k8s.io_v1_clusterrolebinding_capz-manager-rolebinding.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  labels:
    app.giantswarm.io/branch: '[[ .Branch ]]'
    app.giantswarm.io/commit: '[[ .SHA ]]'
    app.kubernetes.io/instance: 'RELEASE-NAME'
    app.kubernetes.io/managed-by: 'Helm'
    app.kubernetes.io/name: cluster-api-provider-azure
    app.kubernetes.io/version: '0.0.1'
    cluster.x-k8s.io/provider: infrastructure-azure
    helm.sh/chart: cluster-api-provider-azure
  name: capz-manager-rolebinding
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: capz-manager-role
subjects:
- kind: ServiceAccount
  name: capz-manager
  namespace: 'giantswarm'
---
# Source: cluster-api-provider-azure/templates/rbac.authorization.k8s.io_v1_role_capz-leader-election-role.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  labels:
    app.giantswarm.io/branch: '[[ .Branch ]]'
    app.giantswarm.io/commit: '[[ .SHA ]]'
    app.kubernetes.io/instance: 'RELEASE-NAME'
    app.kubernetes.io/managed-by: 'Helm'
    app.kubernetes.io/name: cluster-api-provider-azure
    app.kubernetes.io/version: '0.0.1'
    cluster.x-k8s.io/provider: infrastructure-azure
    helm.sh/chart: cluster-api-provider-azure
  name: capz-leader-election-role
  namespace: 'giantswarm'
rules:
- apiGroups:
  - ""
  resources:
  - events
  verbs:
  - create
- apiGroups:
  - coordination.k8s.io
  resources:
  - leases
  verbs:
  - get
  - list
  - watch
  - create
  - update
  - patch
  - delete
---
# Source: cluster-api-provider-azure/templates/rbac.authorization.k8s.io_v1_rolebinding_capz-leader-election-rolebinding.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  labels:
    app.giantswarm.io/branch: '[[ .Branch ]]'
    app.giantswarm.io/commit: '[[ .SHA ]]'
    app.kubernetes.io/instance: 'RELEASE-NAME'
    app.kubernetes.io/managed-by: 'Helm'
    app.kubernetes.io/name: cluster-api-provider-azure
    app.kubernetes.io/version: '0.0.1'
    cluster.x-k8s.io/provider: infrastructure-azure
    helm.sh/chart: cluster-api-provider-azure
  name: capz-leader-election-rolebinding
  namespace: 'giantswarm'
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: Role
  name: capz-leader-election-role
subjects:
- kind: ServiceAccount
  name: capz-manager
  namespace: 'giantswarm'
---
# Source: cluster-api-provider-azure/templates/service.yaml
apiVersion: v1
kind: Service
metadata:
  annotations:
    prometheus.io/scrape: "true"
    giantswarm.io/monitoring: "true"
    giantswarm.io/monitoring-port: "8080"
  labels:
    cluster.x-k8s.io/provider: infrastructure-azure
    app.kubernetes.io/name: "cluster-api-provider-azure"
    app.kubernetes.io/instance: "RELEASE-NAME"
    app.giantswarm.io/branch: "[[ .Branch ]]"
    app.giantswarm.io/commit: "[[ .SHA ]]"
    app.kubernetes.io/managed-by: "Helm"
    app.kubernetes.io/version: "0.0.1"
    helm.sh/chart: "cluster-api-provider-azure-0.0.1"
  name: cluster-api-provider-azure
  namespace: giantswarm
spec:
  ports:
  - name: metrics
    port: 8080
    targetPort: metrics
  - name: webhook
    port: 9443
    targetPort: webhook
  selector:
    cluster.x-k8s.io/provider: infrastructure-azure
    app.kubernetes.io/name: "cluster-api-provider-azure"
    app.kubernetes.io/instance: "RELEASE-NAME"
---
# Source: cluster-api-provider-azure/templates/v1_service_capz-webhook-service.yaml
apiVersion: v1
kind: Service
metadata:
  annotations:
    giantswarm.io/monitoring: "true"
    giantswarm.io/monitoring-port: "8080"
    prometheus.io/scrape: "true"
  labels:
    app.giantswarm.io/branch: '[[ .Branch ]]'
    app.giantswarm.io/commit: '[[ .SHA ]]'
    app.kubernetes.io/instance: 'RELEASE-NAME'
    app.kubernetes.io/managed-by: 'Helm'
    app.kubernetes.io/name: cluster-api-provider-azure
    app.kubernetes.io/version: '0.0.1'
    cluster.x-k8s.io/provider: infrastructure-azure
    helm.sh/chart: cluster-api-provider-azure
  name: capz-webhook-service
  namespace: 'giantswarm'
spec:
  ports:
  - name: metrics
    port: 8080
    targetPort: metrics
  - name: webhook-server
    port: 443
    targetPort: webhook-server
  selector:
    cluster.x-k8s.io/provider: infrastructure-azure
---
# Source: cluster-api-provider-azure/templates/apps_v1_deployment_capz-controller-manager.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    aadpodidbinding: capz-controller-aadpodidentity-selector
    app.giantswarm.io/branch: '[[ .Branch ]]'
    app.giantswarm.io/commit: '[[ .SHA ]]'
    app.kubernetes.io/instance: 'RELEASE-NAME'
    app.kubernetes.io/managed-by: 'Helm'
    app.kubernetes.io/name: cluster-api-provider-azure
    app.kubernetes.io/version: '0.0.1'
    cluster.x-k8s.io/provider: infrastructure-azure
    control-plane: capz-controller-manager
    helm.sh/chart: cluster-api-provider-azure
  name: capz-controller-manager
  namespace: 'giantswarm'
spec:
  replicas: 1
  selector:
    matchLabels:
      cluster.x-k8s.io/provider: infrastructure-azure
      control-plane: capz-controller-manager
  template:
    metadata:
      annotations:
        kubectl.kubernetes.io/default-logs-container: manager
      labels:
        aadpodidbinding: capz-controller-aadpodidentity-selector
        cluster.x-k8s.io/provider: infrastructure-azure
        control-plane: capz-controller-manager
    spec:
      containers:
      - args:
        - --metrics-bind-addr=0.0.0.0:8080
        - --feature-gates=MachinePool=true,AKS=false
        - --watch-filter=capi
        - --v=0
        env:
        - name: AZURE_SUBSCRIPTION_ID
          valueFrom:
            secretKeyRef:
              key: azure.azureoperator.subscriptionid
              name: credential-default
        - name: AZURE_TENANT_ID
          valueFrom:
            secretKeyRef:
              key: azure.azureoperator.tenantid
              name: credential-default
        - name: AZURE_CLIENT_ID
          valueFrom:
            secretKeyRef:
              key: azure.azureoperator.clientid
              name: credential-default
        - name: AZURE_CLIENT_SECRET
          valueFrom:
            secretKeyRef:
              key: azure.azureoperator.clientsecret
              name: credential-default
        - name: NODE_NAME
          valueFrom:
            fieldRef:
              fieldPath: spec.nodeName
        - name: POD_NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
        - name: POD_NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
        image: 'quay.io/giantswarm/cluster-api-azure-controller:v0.5.3'
        imagePullPolicy: IfNotPresent
        livenessProbe:
          httpGet:
            path: /healthz
            port: healthz
          initialDelaySeconds: 10
          periodSeconds: 10
        name: manager
        ports:
        - containerPort: 8080
          name: metrics
          protocol: TCP
        - containerPort: 9443
          name: webhook-server
          protocol: TCP
        - containerPort: 9440
          name: healthz
          protocol: TCP
        readinessProbe:
          httpGet:
            path: /readyz
            port: healthz
          initialDelaySeconds: 10
          periodSeconds: 10
        volumeMounts:
        - mountPath: /tmp/k8s-webhook-server/serving-certs
          name: cert
          readOnly: true
      serviceAccountName: capz-manager
      terminationGracePeriodSeconds: 10
      volumes:
      - name: cert
        secret:
          defaultMode: 420
          secretName: capz-webhook-service-cert
---
# Source: cluster-api-provider-azure/templates/cert-manager.io_v1_certificate_capz-serving-cert.yaml
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  labels:
    app.giantswarm.io/branch: '[[ .Branch ]]'
    app.giantswarm.io/commit: '[[ .SHA ]]'
    app.kubernetes.io/instance: 'RELEASE-NAME'
    app.kubernetes.io/managed-by: 'Helm'
    app.kubernetes.io/name: cluster-api-provider-azure
    app.kubernetes.io/version: '0.0.1'
    cluster.x-k8s.io/provider: infrastructure-azure
    helm.sh/chart: cluster-api-provider-azure
  name: capz-serving-cert
  namespace: 'giantswarm'
spec:
  dnsNames:
  - capz-webhook-service.giantswarm.svc
  - capz-webhook-service.giantswarm.svc.cluster.local
  issuerRef:
    group: cert-manager.io
    kind: ClusterIssuer
    name: selfsigned-giantswarm
  secretName: capz-webhook-service-cert
---
# Source: cluster-api-provider-azure/templates/admissionregistration.k8s.io_v1_mutatingwebhookconfiguration_zzz-capz-mutating-webhook-configuration.yaml
apiVersion: admissionregistration.k8s.io/v1
kind: MutatingWebhookConfiguration
metadata:
  annotations:
    cert-manager.io/inject-ca-from: 'giantswarm/capz-serving-cert'
  labels:
    app.giantswarm.io/branch: '[[ .Branch ]]'
    app.giantswarm.io/commit: '[[ .SHA ]]'
    app.kubernetes.io/instance: 'RELEASE-NAME'
    app.kubernetes.io/managed-by: 'Helm'
    app.kubernetes.io/name: cluster-api-provider-azure
    app.kubernetes.io/version: '0.0.1'
    cluster.x-k8s.io/provider: infrastructure-azure
    helm.sh/chart: cluster-api-provider-azure
  name: zzz-capz-mutating-webhook-configuration
webhooks:
- admissionReviewVersions:
  - v1
  - v1beta1
  clientConfig:
    service:
      name: capz-webhook-service
      namespace: 'giantswarm'
      path: /mutate-infrastructure-cluster-x-k8s-io-v1beta1-azurecluster
  failurePolicy: Fail
  matchPolicy: Equivalent
  name: default.azurecluster.infrastructure.cluster.x-k8s.io
  objectSelector:
    matchLabels:
      cluster.x-k8s.io/watch-filter: capi
  rules:
  - apiGroups:
    - infrastructure.cluster.x-k8s.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - UPDATE
    resources:
    - azureclusters
  sideEffects: None
- admissionReviewVersions:
  - v1
  - v1beta1
  clientConfig:
    service:
      name: capz-webhook-service
      namespace: 'giantswarm'
      path: /mutate-infrastructure-cluster-x-k8s-io-v1beta1-azuremachine
  failurePolicy: Fail
  matchPolicy: Equivalent
  name: default.azuremachine.infrastructure.cluster.x-k8s.io
  objectSelector:
    matchLabels:
      cluster.x-k8s.io/watch-filter: capi
  rules:
  - apiGroups:
    - infrastructure.cluster.x-k8s.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - UPDATE
    resources:
    - azuremachines
  sideEffects: None
- admissionReviewVersions:
  - v1
  - v1beta1
  clientConfig:
    service:
      name: capz-webhook-service
      namespace: 'giantswarm'
      path: /mutate-infrastructure-cluster-x-k8s-io-v1beta1-azuremachinetemplate
  failurePolicy: Fail
  matchPolicy: Equivalent
  name: default.azuremachinetemplate.infrastructure.cluster.x-k8s.io
  objectSelector:
    matchLabels:
      cluster.x-k8s.io/watch-filter: capi
  rules:
  - apiGroups:
    - infrastructure.cluster.x-k8s.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - UPDATE
    resources:
    - azuremachinetemplates
  sideEffects: None
- admissionReviewVersions:
  - v1
  - v1beta1
  clientConfig:
    service:
      name: capz-webhook-service
      namespace: 'giantswarm'
      path: /mutate-infrastructure-cluster-x-k8s-io-v1beta1-azuremachinepool
  failurePolicy: Fail
  name: default.azuremachinepool.infrastructure.cluster.x-k8s.io
  objectSelector:
    matchLabels:
      cluster.x-k8s.io/watch-filter: capi
  rules:
  - apiGroups:
    - infrastructure.cluster.x-k8s.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - UPDATE
    resources:
    - azuremachinepools
  sideEffects: None
- admissionReviewVersions:
  - v1
  - v1beta1
  clientConfig:
    service:
      name: capz-webhook-service
      namespace: 'giantswarm'
      path: /mutate-infrastructure-cluster-x-k8s-io-v1beta1-azuremanagedcontrolplane
  failurePolicy: Fail
  name: default.azuremanagedcontrolplanes.infrastructure.cluster.x-k8s.io
  objectSelector:
    matchLabels:
      cluster.x-k8s.io/watch-filter: capi
  rules:
  - apiGroups:
    - infrastructure.cluster.x-k8s.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - UPDATE
    resources:
    - azuremanagedcontrolplanes
  sideEffects: None
- admissionReviewVersions:
  - v1
  - v1beta1
  clientConfig:
    service:
      name: capz-webhook-service
      namespace: 'giantswarm'
      path: /mutate-infrastructure-cluster-x-k8s-io-v1beta1-azuremanagedmachinepool
  failurePolicy: Fail
  matchPolicy: Equivalent
  name: default.azuremanagedmachinepools.infrastructure.cluster.x-k8s.io
  objectSelector:
    matchLabels:
      cluster.x-k8s.io/watch-filter: capi
  rules:
  - apiGroups:
    - infrastructure.cluster.x-k8s.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - UPDATE
    resources:
    - azuremanagedmachinepools
  sideEffects: None
---
# Source: cluster-api-provider-azure/templates/admissionregistration.k8s.io_v1_validatingwebhookconfiguration_capz-validating-webhook-configuration.yaml
apiVersion: admissionregistration.k8s.io/v1
kind: ValidatingWebhookConfiguration
metadata:
  annotations:
    cert-manager.io/inject-ca-from: 'giantswarm/capz-serving-cert'
  labels:
    app.giantswarm.io/branch: '[[ .Branch ]]'
    app.giantswarm.io/commit: '[[ .SHA ]]'
    app.kubernetes.io/instance: 'RELEASE-NAME'
    app.kubernetes.io/managed-by: 'Helm'
    app.kubernetes.io/name: cluster-api-provider-azure
    app.kubernetes.io/version: '0.0.1'
    cluster.x-k8s.io/provider: infrastructure-azure
    helm.sh/chart: cluster-api-provider-azure
  name: capz-validating-webhook-configuration
webhooks:
- admissionReviewVersions:
  - v1
  - v1beta1
  clientConfig:
    service:
      name: capz-webhook-service
      namespace: 'giantswarm'
      path: /validate-infrastructure-cluster-x-k8s-io-v1beta1-azurecluster
  failurePolicy: Fail
  matchPolicy: Equivalent
  name: validation.azurecluster.infrastructure.cluster.x-k8s.io
  objectSelector:
    matchLabels:
      cluster.x-k8s.io/watch-filter: capi
  rules:
  - apiGroups:
    - infrastructure.cluster.x-k8s.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - UPDATE
    resources:
    - azureclusters
  sideEffects: None
- admissionReviewVersions:
  - v1
  - v1beta1
  clientConfig:
    service:
      name: capz-webhook-service
      namespace: 'giantswarm'
      path: /validate-infrastructure-cluster-x-k8s-io-v1beta1-azuremachine
  failurePolicy: Fail
  matchPolicy: Equivalent
  name: validation.azuremachine.infrastructure.cluster.x-k8s.io
  objectSelector:
    matchLabels:
      cluster.x-k8s.io/watch-filter: capi
  rules:
  - apiGroups:
    - infrastructure.cluster.x-k8s.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - UPDATE
    resources:
    - azuremachines
  sideEffects: None
- admissionReviewVersions:
  - v1
  - v1beta1
  clientConfig:
    service:
      name: capz-webhook-service
      namespace: 'giantswarm'
      path: /validate-infrastructure-cluster-x-k8s-io-v1beta1-azuremachinetemplate
  failurePolicy: Fail
  matchPolicy: Equivalent
  name: validation.azuremachinetemplate.infrastructure.cluster.x-k8s.io
  objectSelector:
    matchLabels:
      cluster.x-k8s.io/watch-filter: capi
  rules:
  - apiGroups:
    - infrastructure.cluster.x-k8s.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - UPDATE
    resources:
    - azuremachinetemplates
  sideEffects: None
- admissionReviewVersions:
  - v1
  - v1beta1
  clientConfig:
    service:
      name: capz-webhook-service
      namespace: 'giantswarm'
      path: /validate-infrastructure-cluster-x-k8s-io-v1beta1-azuremachinepool
  failurePolicy: Fail
  name: validation.azuremachinepool.infrastructure.cluster.x-k8s.io
  objectSelector:
    matchLabels:
      cluster.x-k8s.io/watch-filter: capi
  rules:
  - apiGroups:
    - infrastructure.cluster.x-k8s.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - UPDATE
    resources:
    - azuremachinepools
  sideEffects: None
- admissionReviewVersions:
  - v1
  - v1beta1
  clientConfig:
    service:
      name: capz-webhook-service
      namespace: 'giantswarm'
      path: /validate-infrastructure-cluster-x-k8s-io-v1beta1-azuremachinepoolmachine
  failurePolicy: Fail
  name: azuremachinepoolmachine.kb.io
  objectSelector:
    matchLabels:
      cluster.x-k8s.io/watch-filter: capi
  rules:
  - apiGroups:
    - infrastructure.cluster.x-k8s.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - UPDATE
    resources:
    - azuremachinepoolmachines
  sideEffects: None
- admissionReviewVersions:
  - v1
  - v1beta1
  clientConfig:
    service:
      name: capz-webhook-service
      namespace: 'giantswarm'
      path: /validate-infrastructure-cluster-x-k8s-io-v1beta1-azuremanagedcontrolplane
  failurePolicy: Fail
  name: validation.azuremanagedcontrolplanes.infrastructure.cluster.x-k8s.io
  objectSelector:
    matchLabels:
      cluster.x-k8s.io/watch-filter: capi
  rules:
  - apiGroups:
    - infrastructure.cluster.x-k8s.io
    apiVersions:
    - v1beta1
    operations:
    - CREATE
    - UPDATE
    resources:
    - azuremanagedcontrolplanes
  sideEffects: None
- admissionReviewVersions:
  - v1
  - v1beta1
  clientConfig:
    service:
      name: capz-webhook-service
      namespace: 'giantswarm'
      path: /validate-infrastructure-cluster-x-k8s-io-v1beta1-azuremanagedmachinepool
  failurePolicy: Fail
  matchPolicy: Equivalent
  name: validation.azuremanagedmachinepools.infrastructure.cluster.x-k8s.io
  objectSelector:
    matchLabels:
      cluster.x-k8s.io/watch-filter: capi
  rules:
  - apiGroups:
    - infrastructure.cluster.x-k8s.io
    apiVersions:
    - v1beta1
    operations:
    - UPDATE
    - DELETE
    resources:
    - azuremanagedmachinepools
  sideEffects: None
```

</details>

File: helm/cluster-api-provider-azure/templates/apps_v1_deployment_capz-controller-manager.yaml [hunk 1]
```diff
@@ -64,34 +68,39 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: metadata.namespace
-        image: "{{ .Values.image.registry }}/{{ .Values.image.name }}:{{ .Values.image.tag }}"
-        imagePullPolicy: Always
+        image: '{{ .Values.image.registry }}/{{ .Values.image.name }}:{{ .Values.image.tag }}'
+        imagePullPolicy: IfNotPresent
```
Comment (MarcelMue) [inline helm/cluster-api-provider-azure/templates/apps_v1_deployment_capz-controller-manager.yaml:72]: Why change from Always here?

**Review by njuettner — APPROVED** [review]
lgtm

File: helm/cluster-api-provider-azure/templates/apps_v1_deployment_capz-controller-manager.yaml (hunk 1 above)
Comment (fiunchinho) [reply helm/cluster-api-provider-azure/templates/apps_v1_deployment_capz-controller-manager.yaml:72]: No particular reason, it's just what upstream sets. If you think it's useful having `Always` I can change it back.

Comment (MarcelMue) [reply helm/cluster-api-provider-azure/templates/apps_v1_deployment_capz-controller-manager.yaml:72]: Works for me if upstream sets it like this.

**Review by MarcelMue — APPROVED** [review]

**Pull request merged**